lib/rouge/lexers/lean.rb

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Lean < RegexLexer
+      title "lean"
+      desc 'Lean (leanprover.github.io)'
+      tag 'lean'
+      mimetypes 'text/x-lean'
+
+      keywords = Set.new %w(
```

Comment:
Can you wrap these local variables in memoised methods? So for example:

```ruby
def self.keywords
  @keywords ||= Set.new %w(...)
end
```

This helps a little with overall performance.

Within the respective rules, it means that calls to `keywords` should be replaced with `self.class.keywords`.